pages/profile/[id].tsx

```diff
@@ -0,0 +1,118 @@
+  user: UserProfile;
+  statusCode: number;
+}
+
+const ProfilePage: NextPage<Props> = ({ user, statusCode }: Props) => {
+  function possessiveForm(name: string): string {
+    return name.endsWith('s') ? `${name}'` : `${name}'s`;
+  }
+
+  const isErrorCode = statusCode < 200 || statusCode >= 300;
+  if (isErrorCode || !user) {
+    return <ErrorPage statusCode={isErrorCode ? statusCode : 404}/>;
+  }
+
+  return (
+    <>
+      <Container className="flex flex-col gap-4 pt-20 items-center">
+        <Box component="header">
+          <Typography variant="h1" className="text-5xl font-700 text-center">
+            {possessiveForm(user.discord_name)} Profile
+          </Typography>
+        </Box>
+
+        {user.bio && <Typography variant="body1" component="p">{user.bio}</Typography>}
+
+        <List className="flex gap-2">
+          {user.twitter_username && (
+            <ListItemButton
+              component="a"
+              href={`https://twitter.com/${user.twitter_username}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <TwitterIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.linkedin_url && (
+            <ListItemButton
+              component="a"
+              href={user.linkedin_url}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <LinkedInIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.github_username && (
+            <ListItemButton
+              component="a"
+              href={`https://github.com/${user.github_username}`}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <GitHubIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+
+          {user.website && (
+            <ListItemButton
+              component="a"
+              href={user.website}
+              target="_blank"
+              rel="noopener noreferrer"
+              className="p-2 rounded-full bg-gray-100 hover:bg-gray-200"
+            >
+              <ListItemIcon sx={{ minWidth: 'auto' }}>
+                <HomeRoundedIcon className="text-gray-800"/>
+              </ListItemIcon>
+            </ListItemButton>
+          )}
+        </List>
+      </Container>
+    </>
+  );
+};
+
+ProfilePage.getInitialProps = async ({ req, res, query }: NextPageContext): Promise<Props> => {
```

Comment:
does the return type need to be explicitly defined here?

Reply:
Need to be, no. Should it be? Depends what team you're on. Some say always, some say never, and others say it depends... are you branching? 🤔

Reply:
Fair. The jury is still out for me. I'm leaning towards always as I feel like being explicit is a nice reminder, especially when refactoring. Anyway, I can remove it.